src/App.tsx

```diff
@@ -288,7 +288,11 @@ function App(): ReactElement {
     }
     const setupInitialParameters = async (): Promise<void> => {
       if (initialUrlParams.thresholds) {
-        setFeatureThresholds(initialUrlParams.thresholds);
+        if (dataset) {
+          setFeatureThresholds(validateThresholds(dataset, initialUrlParams.thresholds));
```

Comment:
`validateThresholds` is defined in `data_utils`, and fixes any thresholds that were labeled as categorical that should have been numeric and vice-versa. It's run both on the initial parameter load, which is happening here, and whenever the dataset is changed, which is below.